src/data.jl

```diff
@@ -6,38 +6,47 @@ const API_URL = "https://www.openml.org/api/v1/json"
 # https://www.openml.org/api_docs#!/data/get_data_id
 
 
-"""
-Returns information about a dataset. The information includes the name,
-information about the creator, URL to download it and more.
+function error_msg_handling(e)
+    if isa(e, HTTP.StatusError) && e.status == 412
+        try
+            err = JSON.parse(String(e.response.body))["error"]
+            msg = err["message"]
+            code = err["code"]
+            additional_msg = haskey(err, "additional_message") ? err["additional_message"] : ""
+            @error msg * " " * additional_msg * "(error code $code)"
+        catch
+            @error String(e.response.body)
+        end
+    else
+        println(e)
+    end
+    return nothing
+end
 
-- 110 - Please provide data_id.
-- 111 - Unknown dataset. Data set description with data_id was not found in the database.
-- 112 - No access granted. This dataset is not shared with you.
-"""
-function load_Dataset_Description(id::Int; api_key::String="")
-    url = string(API_URL, "/data/$id")
+function get(url; extra_error_message = "")
     try
-        r = HTTP.request("GET", url)
-        if r.status == 200
-            return JSON.parse(String(r.body))
-        elseif r.status == 110
-            println("Please provide data_id.")
-        elseif r.status == 111
-            println("Unknown dataset. Data set description with data_id was not found in the database.")
-        elseif r.status == 112
-            println("No access granted. This dataset is not shared with you.")
-        end
+        r = HTTP.request("GET", string(API_URL, url))
+        return JSON.parse(String(r.body))
     catch e
-        println("Error occurred. Check if there exists a dataset with id $id.")
-        println("See e.g. OpenML.list_datasets()\n")
-        println(e)
-        return nothing
+        error_msg_handling(e)
+        extra_error_message != "" && println(extra_error_message)
```

Comment:
We never actually throw an exception, only print error messages. Is there a good reason we don't (eventually) throw the exception?

Reply:
No. I fixed this. Thanks.

Reply:
Yes but it looks like this is a premature throw. Don't you want the message at line 32 of data.jl to also be printed?

Reply:
You're right. I changed this now to throw all errors. The error messages look somewhat intimidating, but I guess this is the proper way of doing it.